common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -60,6 +60,24 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
 
   }
 
+  def cleanAmpSoundcloud(document: Document) = {
+    document.getElementsByClass("element-audio").filter { audioElement: Element =>
+      audioElement.getElementsByTag("iframe").length != 0
```

Comment:
you can use isDefined instead of length != 0.  although it's a bit more tricky if it's a java list